lib/nga.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+function handle_error() {
+    echo "Script exited with status ${2} at line ${1}"
+    if [ -z "${savefiles}" ] && [ -d $tmpprefix ]; then rm -f $tmpprefix*; fi
+    if [ -z "${savefiles}" ]  && [ -d $tmpprefix ]; then rm -f $outprefix*; fi
+}
+
+trap 'handle_error ${LINENO} $?' ERR
+
+# cause the script to fail if any commands exist with non-zero status
+set -e
+
+if [[ -z "$1" || -z "$2" || -z "$3" ]]; then
+    echo "Usage:tiff_to_pyramid.bash <tmpdir> <full path to image source> <full path to output target> <save all working files - any value will do>";
+    exit 1;
+fi
+if [[ ! -d $1 || ! -f $2 ]]; then
+    echo "Either the tmpdir or input file do not exist.";
+    exit 1;
+fi
+
+if [ -f $3 ]; then
+    echo "The output file already exists! Deleting it.";
+    if [ -z "${savefiles}" ]; then rm -f $3; fi
+fi
+
+touch $3;
+
+if [ ! -f $3 ]; then
+    echo "Could not create the output file.";
+    exit 1;
+fi
+
+rm -f $3
+
+tmpdir=$1
+# only process the first page of a potentially multi-page image document
+input=$2
+outfile=$3
+savefiles=$4
+pid=$$;
+tmpprefix=${tmpdir}/stripped_srgb_${pid}
+outprefix=${tmpdir}/srgb_${pid}
+
+# cleanup temp files that might already exist
+if [ -z "${savefiles}" ]; then rm -f $tmpprefix*; fi
+if [ -z "${savefiles}" ]; then rm -f $outprefix*; fi
+
+# first, use vipsheader to read the bands
+CHANNELS=$(identify -format "%[channels]\n" ${input}[0] 2>/dev/null)
+echo "channels: ${CHANNELS}"
+if [ ${CHANNELS} = "srgba" ]; then
+    # we have to flatten the image to remove the alpha channel / trasparency before proceeding
+    echo "removing alpha channel from $input"
+    vips im_extract_bands $input ${input}.noalpha.tif 0 3   2>&1
+    if [ -z "${savefiles}" ]; then rm $input; fi
+    mv ${input}.noalpha.tif $input
```

Comment:
Looks like the original file gets overwritten whether or not ${savefiles} is set -- am I misreading this?

Reply:
yeah, lots of fun stuff in this script. it'll also delete itself if temp is an empty string...

Reply:
Let's make sure we cleanly encapsulate these scripts when calling them, but try not to modify them from the NGA originals.